src/crimson/osd/object_context.h

```diff
@@ -164,9 +164,7 @@ class ObjectContext : public Blocker,
     }
   }
   seastar::future<> wait_recovery_read() {
-    return lock.lock_for_read().then([this] {
-      recovery_read_marker = true;
-    });
+    recovery_read_marker = true;
```

Comment:
how about changing the function signature so it returns `void` and update the caller accordingly?

Reply:
oh, yes, sorry, will change that